### DISCUSSION TITLE
如果在局域网搭建，是否支持ECS 或基于地理位置的查询？

### DISCUSSION THREAD

**Discussion by Benjamin1919** [kkkgo/PaoPaoDNS#175] [问答和帮助]
自己已经在软路由上用mosdns做好分流了，因此只想用PaoPaoDNS搭建一个纯粹的递归DNS，即环境变量CNAUTO=no

目前家里宽带有公网IP，软路由也设置好了DDNS，但是看了一下首页的文档，即便是以host网络安装，识别到的也是软路由自身的局域网IP，请问此时的PaoPaoDNS是否支持ECS或基于地理位置的查询呢(根据查询客户端的公网IP给出最优的DNS应答结果)？

后续还想搭配nginx或adguardhome建一个自用的DOH，方便在外地时给自己的手机和电脑用，因此ECS特性对我比较重要。

**Comment by kkkgo** [comment]
参考https://github.com/kkkgo/PaoPaoDNS/issues/8 
ECS一般只用于 终端用户<->公共递归DNS之间有意义（这意味着拥有多条线路出口和多个区域缓存），先不说权威服务器对ECS的支持性，在实际应用中公共递归DNS到权威DNS之间也甚至也不会使用ECS特性（直接使用多个线路请求）

**Reply by Benjamin1919** [reply]
明白～再请教一下，如果软路由上有passwall这种透明代理，某个国内域名的权威DNS在国外，那么查询过程会走代理吗？

**Reply by kkkgo** [reply]
只要你不进行IP层面的分流就不会走代理。文档也提及了“需要注意的是，如果你的网络有“自动分流IP”的功能，请把容器的IP加入不分流的名单，因为权威DNS需要准确的IP去判断，IP分流会影响权威DNS的判断。”
